src/services/codefixes/fixExtendsInterfaceBecomesImplements.ts

```diff
@@ -27,11 +27,22 @@ namespace ts.codefix {
     }
 
     function doChanges(changes: textChanges.ChangeTracker, sourceFile: SourceFile, extendsToken: Node, heritageClauses: ReadonlyArray<HeritageClause>): void {
-        changes.replaceNode(sourceFile, extendsToken, createToken(SyntaxKind.ImplementsKeyword));
+        changes.replaceRange(sourceFile, { pos: extendsToken.getStart(), end: extendsToken.end }, createToken(SyntaxKind.ImplementsKeyword));
         // We replace existing keywords with commas.
         for (let i = 1; i < heritageClauses.length; i++) {
             const keywordToken = heritageClauses[i].getFirstToken()!;

```

Comment:
Might want to assert that this is `SyntaxKind.ImplementsKeyword`.

Reply:
My thinking was that that assertion would fail in invalid code (e.g. multiple extends clauses or an out-of-order extends clause).  I'm not convinced it's worthwhile to harden a best-effort code fix against that sort of bad code (the user will already be getting a diagnostic and the change is undoable), but I don't feel strongly about it.  However, I'd only add the assert if we were also adding actual checks.